src/async_impl/body.rs

```diff
@@ -0,0 +1,98 @@
+use futures::{Stream, Poll, Async};
+use bytes::Bytes;
+
+pub struct Body {
+    inner: Inner,
+}
+
+enum Inner {
+    Reusable(Bytes),
+    Hyper(::hyper::Body),
+}
+
+impl Body {
+    fn poll_inner(&mut self) -> &mut ::hyper::Body {
+        match self.inner {
+            Inner::Hyper(ref mut body) => body,
+            Inner::Reusable(_) => unreachable!(),
+        }
+    }
+}
+
+impl Stream for Body {
+    type Item = Chunk;
+    type Error = ::Error;
+
+    #[inline]
+    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
+        match try_!(self.poll_inner().poll()) {
+            Async::Ready(opt) => Ok(Async::Ready(opt.map(|chunk| Chunk {
+                inner: chunk,
+            }))),
+            Async::NotReady => Ok(Async::NotReady),
+        }
+    }
+}
+
+
+#[derive(Debug, Default)]
+pub struct Chunk {
+    inner: ::hyper::Chunk,
+}
+
+impl ::std::ops::Deref for Chunk {
+    type Target = [u8];
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        self.inner.as_ref()
+    }
+}
+
+impl Extend<u8> for Chunk {
+    fn extend<T>(&mut self, iter: T)
+    where T: IntoIterator<Item=u8> {
+        self.inner.extend(iter)
+    }
+}
+
+impl IntoIterator for Chunk {
+    type Item = u8;
+    //XXX: exposing type from hyper!
+    type IntoIter = <::hyper::Chunk as IntoIterator>::IntoIter;
+    fn into_iter(self) -> Self::IntoIter {
+        self.inner.into_iter()
+    }
+}
+
+// pub(crate)
```

Comment:
On most recent version. It doesn't really cost anything to still compile on the previous version.